### PR TITLE
tests: Helm upgrade integration test

### PR DESCRIPTION
**Pull request by travisn**
**Description of your changes:**
The helm tests were previously only for new installs, and did not have an upgrade path. Now the upgrade path is tested to give confidence in the helm upgrades.

At the same time, include some test code cleanup... The test suites were triggered by Jenkins potentially using different versions of K8s and environment variables. There is no longer a need for these environment vars to determine which test suites should be run. The test suites to run can just be triggered directly.

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: .github/workflows/integration-test-upgrade-suite.yaml [hunk 1]
```diff
@@ -30,18 +30,66 @@ jobs:
...
+      with:
+        fetch-depth: 0
+
+    - name: setup cluster resources
+      uses: ./.github/workflows/integration-test-setup
+      with:
+        github-token: ${{ secrets.GITHUB_TOKEN }}
+        kubernetes-version: ${{ matrix.kubernetes-versions }}
+
+    - name: TestHelmUpgrade
```
Comment (BlaineEXE) [inline .github/workflows/integration-test-upgrade-suite.yaml:78]: I see this is named with "Suite" for the on-release tests. Is there a reason to not keep the names consistent?

```suggestion
    - name: TestHelmUpgradeSuite
```

Comment (travisn) [reply .github/workflows/integration-test-upgrade-suite.yaml:78]: The upgrade tests don't really run as a suite, the github actions call individual tests inside the suite for the ceph upgrade, helm upgrade, and the daily tests for ceph pacific-devel upgrade, etc. So it seemed appropriate to remove the "Suite" suffix from the test name.

File: .github/workflows/integration-tests-on-release.yaml [hunk 2]
```diff
@@ -155,14 +155,56 @@ jobs:
...
+        tests/scripts/collect-logs.sh
+
+    - name: Artifact
+      uses: actions/upload-artifact@v2
+      if: failure()
+      with:
+        name: ceph-upgrade-suite-artifact-${{ matrix.kubernetes-versions }}
+        path: /home/runner/work/rook/rook/tests/integration/_output/tests/
+
+  TestHelmUpgradeSuite:
```
Comment (BlaineEXE) [inline .github/workflows/integration-tests-on-release.yaml:175]: This one ends in `Suite`.

File: .github/workflows/integration-test-upgrade-suite.yaml [hunk 3]
```diff
@@ -30,18 +30,66 @@ jobs:
...
```
Comment (BlaineEXE) [inline .github/workflows/integration-test-upgrade-suite.yaml:78]: This one does not end in `Suite`

File: .github/workflows/integration-test-upgrade-suite.yaml (hunk 1 above)
Comment (BlaineEXE) [reply .github/workflows/integration-test-upgrade-suite.yaml:78]: Right, but one ends in Suite and the other doesn't, and I'm not sure if I've missed something or if one of them should be changed.

File: .github/workflows/integration-test-upgrade-suite.yaml (hunk 3 above)
Comment (travisn) [reply .github/workflows/integration-test-upgrade-suite.yaml:78]: I'll just add Suite back for consistency

**Review by BlaineEXE — APPROVED** [review]

**Pull request merged**